src/objects/Skeleton.js

```diff
@@ -170,6 +150,106 @@ Object.assign( Skeleton.prototype, {
 
 		return undefined;
 
+	},
+
+	init: function () {
+
+		var bones = this.bones;
+		var boneInverses = this.boneInverses;
+
+		this.boneMatrices = new Float32Array( bones.length * 16 );
+
+		// calculate inverse bone matrices if necessary
+
+		if ( boneInverses === undefined ) {
+
+			this.calculateInverses();
+
+		} else {
+
+			// handle special case
+
+			if ( bones.length !== boneInverses.length ) {
+
+				console.warn( 'THREE.Skeleton: Amount of inverse bone matrices does not match amount of bones.' );
+
+				this.boneInverses = [];
+
+				for ( var i = 0, il = this.bones.length; i < il; i ++ ) {
+
+					this.boneInverses.push( new Matrix4() );
+
+				}
+
+			}
+
+		}
+
+		return this;
+
+	},
+
+	fromJSON: function ( json, object ) {
+
+		this.uuid = json.uuid;
+
+		this.boneInverses = [];
+
+		var bones = this.bones;
+		var boneInverses = this.boneInverses;
+
+		for ( var i = 0, l = json.bones.length; i < l; i ++ ) {
+
+			var boneUUID = json.bones[ i ];
+			var bone = object.getObjectByProperty( 'uuid', boneUUID );
```

Comment:
Here can be slow if scene graph is big. Probably making bone lookup table with bone uuid in `parseObject` is good. I think it may be ok with optimizing after we receive the performance improvement request tho. Adding a comment about performance might be good so far?

Reply:
Sounds good 👍